src/skin/qml/asyncimageprovider.cpp

```diff
@@ -0,0 +1,55 @@
+#include "skin/qml/asyncimageprovider.h"
+
+#include "library/coverartcache.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+AsyncImageResponse::AsyncImageResponse(const QString& id, const QSize& requestedSize)
+        : m_id(id), m_requestedSize(requestedSize) {
+    setAutoDelete(false);
+}
+
+QQuickTextureFactory* AsyncImageResponse::textureFactory() const {
+    return QQuickTextureFactory::textureFactoryForImage(m_image);
+}
+
+void AsyncImageResponse::run() {
+    if (m_id.startsWith(QStringLiteral("coverart/"))) {
+        QString trackLocation = QString::fromUtf8(QByteArray::fromBase64(
+                m_id.mid(9).toLatin1(), QByteArray::Base64UrlEncoding));
+
+        // TODO: This code does not allow to override embedded cover art with
```

Comment:
With the track location at hand we could load the track object and grab the actual CoverInfo. Why do we need this hack then?

Using TrackCollectionManager::getTrackByRef(TrackRef::fromFilePath(trackLocation)) should work.